### PR TITLE
Refactor duplicated SSH code

### LINKED ISSUE #85: Refactor duplicated SSH code
_As a developer
I want code that is duplicated by code present in [hcommon-hive-metastore](https://github.com/HotelsDotCom/hcommon-hive-metastore) removed from CircusTrain
So that I have less code to maintain, can fix bugs in one place etc._

**Acceptance Criteria**

* Replace `com.hotels.bdp.circustrain.api.conf.MetastoreTunnel` with `com.hotels.hcommon.hive.metastore.client.tunnelling.MetastoreTunnel`
* Remove `com.hotels.bdp.circustrain.api.validation.constraints.TunnelRoute` and `com.hotels.bdp.circustrain.api.validation.constraintvalidators.TunnelRouteValidator`

Above code tested out in our development environment in AWS via a tunnel to ensure tunnels still work correctly.

### PR DESCRIPTION
**Pull request by AnanaMJ**
Fixes #85

### CODE REVIEW

File: circus-train-api/src/main/java/com/hotels/bdp/circustrain/api/validation/constraints/TunnelRoute.java [hunk 1]
```diff
@@ -1,84 +0,0 @@
...
-
-  /**
-   * Defines several {@code @TunnelHops} annotations on the same element.
-   */
-  @Target({ METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER })
-  @Retention(RUNTIME)
-  @Documented
-  public @interface List {
-    TunnelRoute[] value();
-  }
```
Comment (AnanaMJ) [inline circus-train-api/src/main/java/com/hotels/bdp/circustrain/api/validation/constraints/TunnelRoute.java]: The code between lines 75 and 83 is not in hcommon-hive-metastore, but I could not find anything that uses this functionality.

File: circus-train-api/src/main/java/com/hotels/bdp/circustrain/api/validation/constraints/TunnelRoute.java [hunk 2]
```diff
@@ -1,84 +0,0 @@
...
- * @see TunnelRouteValidator
- */
-@Documented
-@Constraint(validatedBy = { TunnelRouteValidator.class })
-@Target({ METHOD, FIELD, ANNOTATION_TYPE, CONSTRUCTOR, PARAMETER })
-@Retention(RUNTIME)
-@ReportAsSingleViolation
-@Pattern(regexp = "")
-public @interface TunnelRoute {
-  String message() default "{com.hotels.bdp.circustrain.validation.constraints.TunnelHops.message}";
```
Comment (patduin) [inline circus-train-api/src/main/java/com/hotels/bdp/circustrain/api/validation/constraints/TunnelRoute.java]: This reminds me: there is a file in src/main/resources/ValidationMessages.properties. That's where this is referring to, you can delete it too (I suspect it's also in your WD pr)

Comment (AnanaMJ) [reply circus-train-api/src/main/java/com/hotels/bdp/circustrain/api/validation/constraints/TunnelRoute.java]: I can find the one for WD (I've just deleted it) but I can't find the one for this project.

Comment (massdosage) [reply circus-train-api/src/main/java/com/hotels/bdp/circustrain/api/validation/constraints/TunnelRoute.java]: Nothing to delete then ;) I always wondered what was used to resolve this property - now I know! This was complete overkill for our purposes IMHO, we can just hardcode a generic error message for this (which is what we do now in `hcommon-ssh`).

File: circus-train-api/src/main/java/com/hotels/bdp/circustrain/api/validation/constraints/TunnelRoute.java (hunk 1 above)
Comment (AnanaMJ) [reply circus-train-api/src/main/java/com/hotels/bdp/circustrain/api/validation/constraints/TunnelRoute.java]: After removing the code between those lines, the error changes from
```
Errors found in the provided configuration file:
	{com.hotels.bdp.circustrain.validation.constraints.TunnelHops.message}
```
to
```
Errors found in the provided configuration file:
	Invalid SSH tunnel route
```

**Pull request merged**